e2e/tests/regression/test_replica.robot

```diff
@@ -69,4 +69,43 @@ Test Offline Replica Rebuilding
     And Wait until volume 0 replica rebuilding started on node 0
     And Wait for volume 0 detached
     And Volume 0 should have 3 replicas when detached
-    And Set setting offline-replica-rebuilding to false
\ No newline at end of file
+    And Set setting offline-replica-rebuilding to false
+
+Test Preempt Offline Replica Rebuilding By A Workload
+    [Tags]    coretest    offline-rebuilding
+    [Documentation]    Test preempt offline replica rebuilding by a workload.
+    ...
+    ...                Issue: https://github.com/longhorn/longhorn/issues/8443
+    ...    1. Create a workload with Longhorn volume having 3 replicas in a 3 worker nodes cluster
+    ...    2. Write some data to the volume.
+    ...    3. Scale down the workload to detach the volume.
+    ...    4. Delete a replica of the volume.
+    ...    5. Enable the offline rebuilding by the API `volume.offlineReplicaRebuilding`.
+    ...    6. When the rebuilding is in progress, scale up the workload.
+    ...    7. Check if the offline rebuilding process is canceled.
+    ...    8. Check if the workload is working well.
+    Given Create volume 0 with    size=6Gi    numberOfReplicas=3    dataEngine=${DATA_ENGINE}
+    And Attach volume 0
+    And Wait for volume 0 healthy
+    Then Detach volume 0
+    And Wait for volume 0 detached
+    Then Create deployment 0 with volume 0
+    And Wait for volume 0 healthy
+    Then Write 5120 MB data to file data.txt in deployment 0
+    And Scale down deployment 0 to detach volume
+    And Wait for volume 0 detached
+
+    # enable offline replica rebuilding.
+    When Delete volume 0 replica on node 0
+    Then Enable volume 0 offline replica rebuilding
+    # wait for offline rebuilding to start.
+    And Wait until volume 0 replica rebuilding started on node 0
+    And Volume 0 should have 3 replicas running
+    # scale up the workload and it preempts the offline rebuilding.
+    And Scale up deployment 0 to attach volume
+    # the volume might be detched or not so we only wait for volume becomes healthy.
+    # we should try to check if the volume becomes healthy without offline rebuilding.
```

Comment:
I think we should indicate that `online rebuilding takes over`.

Reply:
Added. Thanks.